topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -1151,29 +1272,25 @@ public void handle(Future<Void> fut) {
             });
     }
 
-    private Future<Void> getKafkaAndReconcile(LogContext logContext, TopicName topicName, Topic privateTopic, KafkaTopic kafkaTopicResource) {
+    private Future<Void> getKafkaAndReconcile(Reconciliation reconciliation, LogContext logContext, TopicName topicName,
+                                              Topic privateTopic, KafkaTopic kafkaTopicResource) {
         logContext.withKubeTopic(kafkaTopicResource);
         Future<Void> topicFuture = Future.future();
         try {
             Topic k8sTopic = kafkaTopicResource != null ? TopicSerialization.fromTopicResource(kafkaTopicResource) : null;
-            kafka.topicMetadata(topicName).setHandler(metadataResult -> {
-                if (metadataResult.succeeded()) {
-                    TopicMetadata kafkaTopicMeta = metadataResult.result();
+            kafka.topicMetadata(topicName)
+                .compose(kafkaTopicMeta -> {
                     Topic topicFromKafka = TopicSerialization.fromTopicMetadata(kafkaTopicMeta);
-                    reconcile(logContext, kafkaTopicResource, k8sTopic, topicFromKafka, privateTopic, reconcileResult -> {
-                        if (reconcileResult.succeeded()) {
-                            LOGGER.info("Success reconciling KafkaTopic {}", logTopic(kafkaTopicResource));
-                            topicFuture.complete();
-                        } else {
-                            LOGGER.error("Error reconciling KafkaTopic {}", logTopic(kafkaTopicResource), reconcileResult.cause());
-                            topicFuture.fail(reconcileResult.cause());
-                        }
-                    });
-                } else {
-                    LOGGER.error("Error reconciling KafkaTopic {}", logTopic(kafkaTopicResource), metadataResult.cause());
-                    topicFuture.fail(metadataResult.cause());
-                }
-            });
+                    return reconcile(reconciliation, logContext, kafkaTopicResource, k8sTopic, topicFromKafka, privateTopic);
+                })
+                .setHandler(ar -> {
+                    if (ar.failed()) {
+                        LOGGER.error("Error reconciling KafkaTopic {}", logTopic(kafkaTopicResource), ar.cause());
```

Comment:
Do you miss second `{}` in the string part?

Reply:
No this is an overload of the `error` method which gets a `Throwable` parameter (so `ar.cause`) to print exception information and you don't need the placeholder in the message for that.

Reply:
Aha, good to know! :)